imod_coupler/drivers/ribametamod/ribametamod.py

```diff
@@ -126,75 +145,176 @@ def couple(self) -> None:
+            self.mf6_river_packages,
+            self.mf6_drainage_packages,
+            {"ribasim_nbound": len(self.ribasim_level)},
+        )
+        # MetaSWAP - MODFLOW 6
+        mswmod_packages: Dict[str, Any] = {}
+        mswmod_packages["msw_head"] = self.msw_head
+        mswmod_packages["msw_volume"] = self.msw_volume
+        mswmod_packages["msw_storage"] = self.msw_storage
+        mswmod_packages[
+            "mf6_recharge"
+        ] = self.mf6_recharge  # waar komt mf6_recharge vandaan
+        if self.coupling.enable_sprinkling:
+            mf6_sprinkling_tag = self.mf6.get_var_address(
+                "BOUND", self.coupling.mf6_model, self.coupling.mf6_msw_well_pkg
+            )
+            self.mf6_sprinkling_wells = self.mf6.get_value_ptr(mf6_sprinkling_tag)[:, 0]
+            mswmod_packages["mf6_sprinkling_wells"] = self.mf6_sprinkling_wells
+        mswmod_packages["mf6_head"] = self.mf6_head
+        mswmod_packages["mf6_storage"] = self.mf6_storage
+        mswmod_packages["mf6_has_sc1"] = self.mf6_has_sc1
+        mswmod_packages["mf6_area"] = self.mf6_area
+        mswmod_packages["mf6_top"] = self.mf6_top
+        mswmod_packages["mf6_bot"] = self.mf6_bot
+        self.mapping = SetMapping(
+            self.coupling,
+            ChainMap(
+                ribmod_packages,
+                mswmod_packages,
+            ),
+            self.msw.working_directory / "mod2svat.inp",
+        )
 
     def update(self) -> None:
         # TODO: Store a copy of the river bottom and the river elevation. The
         # river bottom and drainage elevation should not be fall below these
         # values. Note that the river bottom and the drainage elevation may be
         # update every stress period.
-        #
-        # iMOD Python sets MODFLOW 6' time unit to days
-        # Ribasim's time unit is always seconds
-        ribamod_time_factor = 86400
 
-        # Set the MODFLOW 6 river stage and drainage to value of waterlevel of Ribasim basin
-        for key, river in self.mf6_active_river_packages.items():
-            # TODO: use specific level after Ribasim can export levels
-            river.stage[:] = self.map_rib2mod[key].dot(self.ribasim_level)
-        for key, drainage in self.mf6_active_drainage_packages.items():
-            # TODO: use specific level after Ribasim can export levels
-            drainage.elevation[:] = self.map_rib2mod[key].dot(self.ribasim_level)
+        # exchange stages from Ribasim to MODFLOW
+        self.exchange_mod2rib()
+
+        # Do one MODFLOW 6 - MetaSWAP timestep
+        self.update_MODFLOW6_MetaSWAP()
 
-        # One time step in MODFLOW 6
-        self.mf6.update()
+        # exchange drainage fluxes from MODFLOW 6 to Ribasim
+        self.exchange_rib2mod()
 
+        # Update Ribasim until current time of MODFLOW 6
+        self.ribasim.update_until(self.get_current_time() * days_to_seconds(self.delt))
+
+    def update_MODFLOW6_MetaSWAP(self) -> None:
```

Comment:
```suggestion
    def update_modflow6_metaswap(self) -> None:
```

Let's try to stick to snake case for method names: https://developer.mozilla.org/en-US/docs/Glossary/Snake_case